File: 4_sem/ADS/8.py
s = list(input())
l = len(s)

pol = True
for i in range(1,l):
    if (s[0] != s[i]):
        pol = False
        break

if pol:
    print(-1)
else:
    pol = True
    for i in range(l//2):
        if (s[i] != s[l-1-i]):
            pol = False
            break

    if pol:
        print(l-1)
    else:
        print(l)

File: 4_sem/ADS/8.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    string s;
    cin >> s;
    int l = s.length();
    
    bool pol = true;
    for (int i = 1; i < l; i++)
    {
        if (s[0] != s[i])
        {
            pol = false;
            break;
        }
    }

    if (pol)
    {
        cout << -1;
        return 0;
    }

    pol = true;
    for (int i = 0; i < l/2; i++)
    {
        if (s[i] != s[l-1-i])
        {
            pol = false;
            break;
        }
    }

    if (pol)
    {
        cout << l - 1;
    }
    else
    {
        cout << l;
    }

    return 0;
}